qiskit/synthesis/two_qubit/two_qubit_decompose.py

```diff
@@ -280,157 +279,29 @@ def __init__(self, rxx_equivalent_gate: Type[Gate]):
-                raise QiskitError("Equivalent gate needs to take exactly 1 angle parameter.") from _
-            decomp = TwoQubitWeylDecomposition(rxx_equivalent_gate(test_angle))
-
-            circ = QuantumCircuit(2)
-            circ.rxx(test_angle, 0, 1)
-            decomposer_rxx = TwoQubitWeylDecomposition(
-                Operator(circ).data,
-                fidelity=None,
-                _specialization=two_qubit_decompose.Specialization.ControlledEquiv,
-            )
-
-            circ = QuantumCircuit(2)
-            circ.append(rxx_equivalent_gate(test_angle), qargs=[0, 1])
-            decomposer_equiv = TwoQubitWeylDecomposition(
-                Operator(circ).data,
-                fidelity=None,
-                _specialization=two_qubit_decompose.Specialization.ControlledEquiv,
-            )
-
-            scale = decomposer_rxx.a / decomposer_equiv.a
-
-            if abs(decomp.a * 2 - test_angle / scale) > atol:
-                raise QiskitError(
-                    f"{rxx_equivalent_gate.__name__} is not equivalent to an RXXGate."
-                )
-
-            scales.append(scale)
-
-        # Check that all three tested angles give the same scale
-        if not np.allclose(scales, [scale] * len(test_angles)):
-            raise QiskitError(
-                f"Cannot initialize {self.__class__.__name__}: with gate {rxx_equivalent_gate}. "
-                "Inconsistent scaling parameters in checks."
-            )
-
-        self.scale = scales[0]
-
-        self.rxx_equivalent_gate = rxx_equivalent_gate
-
-    def __call__(self, unitary, *, atol=DEFAULT_ATOL) -> QuantumCircuit:
-        """Returns the Weyl decomposition in circuit form.
-
-        Note: atol ist passed to OneQubitEulerDecomposer.
-        """
-
-        # pylint: disable=attribute-defined-outside-init
-        self.decomposer = TwoQubitWeylDecomposition(unitary)
-
-        oneq_decompose = OneQubitEulerDecomposer("ZYZ")
-        c1l, c1r, c2l, c2r = (
-            oneq_decompose(k, atol=atol)
-            for k in (
-                self.decomposer.K1l,
-                self.decomposer.K1r,
-                self.decomposer.K2l,
-                self.decomposer.K2r,
-            )
+        self._inner_decomposition = two_qubit_decompose.TwoQubitControlledUDecomposer(
+            rxx_equivalent_gate._standard_gate
         )
-        circ = QuantumCircuit(2, global_phase=self.decomposer.global_phase)
-        circ.compose(c2r, [0], inplace=True)
-        circ.compose(c2l, [1], inplace=True)
-        self._weyl_gate(circ)
-        circ.compose(c1r, [0], inplace=True)
-        circ.compose(c1l, [1], inplace=True)
-        return circ
-
-    def _to_rxx_gate(self, angle: float) -> QuantumCircuit:
-        """
-        Takes an angle and returns the circuit equivalent to an RXXGate with the
-        RXX equivalent gate as the two-qubit unitary.
+        self.rxx_equivalent_gate = self._inner_decomposition.rxx_equivalent_gate
```

Comment:
This makes a second copy of the inner rust gate object which won't be the most useful from python. I would probably store the python gate instead:

```suggestion
        self.rxx_equivalent_gate = rxx_equivalent_gate
```

Reply:
done in [e532c8a](https://github.com/Qiskit/qiskit/pull/13139/commits/e532c8a9e1df7dff37331b6413857c1f53a4ce36)